### PR TITLE
Create a new target

### PR DESCRIPTION
**Pull request by kevquincke**
#### Trello board reference:

* https://trello.com/c/7pRl62ww/4-3-as-a-user-i-should-be-able-to-create-a-new-target
* https://trello.com/c/YDXkMDai/6-2-as-a-user-i-should-only-be-able-to-create-a-maximum-of-10-targets

---

#### Description:

* As a user I should be able to create a new target.
* As a user I should only be able to create a maximum of 10 targets

### Screenshots:
image from ios 1
image from ios 2
image from ios 3
image from ios

### CODE REVIEW

File: src/components/CreateTargetForm/index.js [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+        error={error.topic}
+      />
+
+      <CustomButton
+        onPress={onCreateTarget}
+        title="Save Target"
+        capitalized
+        style={createTargetButton}
+      />
+    </View>
```
Comment (missx) [inline src/components/CreateTargetForm/index.js]: Can you still see the inputs in this form when you add a value and the keyboard is displayed?

File: src/containers/TargetPointsScreen/index.js [hunk 2]
```diff
@@ -53,33 +88,215 @@ class TargetPointsScreen extends Component {
...
+        title: null,
+        latitude: 0,
+        longitude: 0,
+        topic: -1
+      }
+    });
+  }
+
+  onCreateTarget() {
+    console.log('entraaaaa');
```
Comment (missx) [inline src/containers/TargetPointsScreen/index.js]: Hahahah remove this line. :)

File: src/components/CreateTargetForm/index.js (hunk 1 above)
Comment (kevquincke) [reply src/components/CreateTargetForm/index.js]: Yes, I used KeyboardAvoidingView component in order to do that, but it's being used from outside. Now that you mention that, maybe I can move that component to the form and replace the View with that.

File: src/containers/TargetPointsScreen/index.js (hunk 2 above)
Comment (kevquincke) [reply src/containers/TargetPointsScreen/index.js]: Ohhhh shoot! haha, I forgot to remove that line

**Comment by kevquincke** [comment]
@missx KeyboardAvoidingView moved to create target form, console.log removed, and some issues with focus and blur on the form fixed

File: src/components/CreateTargetForm/index.js [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+        label="SELECT A TOPIC"
+        values={topicValues}
+        selectedValue={topic}
+        onFocus={onTitleBlur}
+        placeholder="What do you want to talk about?"
+        onChange={value => onFieldChange('topic', value)}
+        style={input}
+        error={error.topic}
+      />
+
```
Comment (aguscha333) [inline src/components/CreateTargetForm/index.js]: delete extra line

File: src/components/CreateTargetForm/index.js [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+      />
+    </KeyboardAvoidingView>
+  );
+};
+
+CreateTargetForm.propTypes = {
+  onFieldChange: func.isRequired,
+  onCreateTarget: func.isRequired,
+  onCancelCreateTarget: func.isRequired,
+  onTitleFocus: func,
```
Comment (aguscha333) [inline src/components/CreateTargetForm/index.js]: is this not required? same about the one below

File: src/containers/TargetPointsScreen/index.js [hunk 5]
```diff
@@ -53,33 +88,212 @@ class TargetPointsScreen extends Component {
...
+      );
+    }
+  }
+
+  onCancelCreateNewTarget() {
+    this.setState({
+      showCreateNewTargetModal: false,
+      customMapStyle: styles.mapWithoutForm,
+      newTargetValues: {
+        areaLength: 0,
```
Comment (aguscha333) [inline src/containers/TargetPointsScreen/index.js]: this properties are being repeated below, maybe it is a good idea to define it in a constant. In this file or in the constants file, wherever you see fit.

File: src/actions/targetActions.js [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+
+export const getTargets = () =>
+  (dispatch) => {
+    dispatch(startRequest());
+
+    return api.getTargets().then(({ targets }) => {
+      dispatch(finishRequest());
+      dispatch(getTargetSuccess(targets));
+    }).catch((err) => {
+      throw err.errors;
```
[the remark this answers is not in the document]
Comment (kevquincke) [reply src/actions/targetActions.js]: You're totally right, I forgot to do that

**Comment by kevquincke** [comment]
@aguscha333 all issues have been resolved

**Pull request merged**